platform-docs/docs/intro.md

```diff
@@ -30,13 +30,39 @@ To build a block editor, you need to install the following dependencies:
 
 We're going to be using JSX to write our UI and components. So one of the first steps we need to do is to configure our build tooling, By default vite supports JSX and and outputs the result as a React pragma. The Block editor uses React so there's no need to configure anything here but if you're using a different bundler/build tool, make sure the JSX transpilation is setup properly.
 
+## Side note: `process.env.IS_GUTENBERG_PLUGIN`
```

Comment:
I think we should fix this in Gutenberg directly. We should make sure that it works by default without setting this rather than having npm consumers learn about this.

Reply:
yeah, that makes sense! I see that this is actually a GB issue because it should have been replaced at build time and we even [have a lint for it](https://github.com/WordPress/gutenberg/blob/ae904142f1c4c47c3d83ecc698f9e71da769bf1d/packages/eslint-plugin/docs/rules/is-gutenberg-plugin.md#L1-L2).

Do you have any idea why that would not be the case?

Reply:
I see that the version of the packages published to npm does not have the value of `process.env.IS_GUTENBERG_PLUGIN` replaced. E.g.: You can check here: https://unpkg.com/browse/@wordpress/core-data@6.31.0/build/resolvers.js

Reply:
@michalczaplinski yes, that's fine, because Gutenberg also consumes the built npm packages like any other app. The problem is that it shouldn't cause an error by default but should assume "false" if you don't define the variable.